sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1257,7 +1257,8 @@ class DataFrame private[sql](
   def drop(colNames: String*): DataFrame = {
     val resolver = sqlContext.analyzer.resolver
     val remainingCols =
-      schema.filter(f => colNames.forall(n => !resolver(f.name, n))).map(f => Column(f.name))
+      schema.filter(f => colNames.forall(n => !resolver(f.name, n)))
```

Comment:
super nit: i would consider breaking before the filter also if you have to break for the map.  otherwise the filter looks special when really you are just chaining a bunch of operations.